
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>A Denoise Algorithm &#8212; Inference for Gene Regulartory Networks</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.acff12b8f9c144ce68a297486a2fa670.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/style.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-svg.js"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="GRGNN" href="71-grgnn.html" />
    <link rel="prev" title="Sup, semi-sup, unsup inference" href="31-grn-ssu.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      
      <h1 class="site-logo" id="site-title">Inference for Gene Regulartory Networks</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../index.html">
   Inference for Gene Regulartory Networks
  </a>
 </li>
</ul>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../intro/00-intro.html">
   Introduction
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../intro/11-grn.html">
     Gene Regulatory Networks
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="00-summary.html">
   Paper Summary
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="11-bio-net-dl.html">
     Biological networks and deep learning
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="31-grn-ssu.html">
     Sup, semi-sup, unsup inference
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     A Denoise Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="71-grgnn.html">
     GRGNN
    </a>
   </li>
  </ul>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/summary/53-grn-denoise.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#abstract">
   Abstract
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#related-work">
   Related Work
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#unsupservised-methods">
     Unsupservised Methods
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#supervised-methods">
     Supervised Methods
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#method">
   Method
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#feature-vector-construction">
     Feature Vector Construction
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-clearning-algorithm">
     Data Clearning Algorithm
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experiments">
   Experiments
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data">
     Data
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#training-set">
       Training set
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#test-set">
       Test set
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#pipeline">
     Pipeline
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#evaluation">
     Evaluation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#results">
     Results
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#takeaway">
   Takeaway
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#ref-of-interest">
     Ref of interest
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#questions">
     Questions
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="a-denoise-algorithm">
<h1>A Denoise Algorithm<a class="headerlink" href="#a-denoise-algorithm" title="Permalink to this headline">¶</a></h1>
<p>[<a class="reference external" href="https://web.njit.edu/~wangj/publications/ARTICLES/ICMLA2016.pdf">PDF</a>] Turki, Turki, Jason TL Wang, and Ibrahim Rajikhan. “Inferring gene regulatory networks by combining supervised and unsupervised methods.” In 2016 15th IEEE International Conference on Machine Learning and Applications (ICMLA), pp. 140-145. IEEE, 2016.</p>
<p>One-line summary</p>
<ul class="simple">
<li><p>they develop a denoise algorithm on gene expression data before fitting the data to supervised algorithm that improves the classification performance.</p></li>
</ul>
<div class="section" id="abstract">
<h2>Abstract<a class="headerlink" href="#abstract" title="Permalink to this headline">¶</a></h2>
<p>Motivation</p>
<ul class="simple">
<li><p>Supervised methods for inferring gene regulatory networks (GRNs) perform well with good training data. However, when training data is absent, these methods are not applicable.</p></li>
<li><p>Unsupervised methods do not need training data but their accuracy is low.</p></li>
<li><p>In this paper, we combine them.</p></li>
</ul>
<p>Hightlight</p>
<ul class="simple">
<li><p>We use results obtained from unsupervised methods to train supervised methods.</p></li>
<li><p>Since the results contain noise, we develop a <strong>data cleaning algorithm</strong> to remove noise, hence improving the quality of the training data.</p></li>
<li><p>These refined training data are then used to guide classifiers, e.g. SVM, NN for link prediction</p></li>
</ul>
</div>
<div class="section" id="related-work">
<h2>Related Work<a class="headerlink" href="#related-work" title="Permalink to this headline">¶</a></h2>
<div class="section" id="unsupservised-methods">
<h3>Unsupservised Methods<a class="headerlink" href="#unsupservised-methods" title="Permalink to this headline">¶</a></h3>
<p>BANJO [11]: Bayesian Network Inference with Jave Objects</p>
<ul class="simple">
<li><p>models GRNs as a first-order Markov process</p></li>
<li><p>searches through all possible GRNs for the one with the best score</p></li>
</ul>
<p>TimeDelay-ARACNE [12]</p>
<ul class="simple">
<li><p>use mutual information</p></li>
</ul>
<p>tlCLR [13, 14]</p>
<ul class="simple">
<li><p>use mutual information and ODE</p></li>
</ul>
<p>DFG [15]: Dynamic Factor Graphs</p>
<ul class="simple">
<li><p>model experimental noise as a fitted noise</p></li>
</ul>
<p>Jump3 [16]</p>
<ul class="simple">
<li><p>noon-parametric, based on decision trees</p></li>
</ul>
<p>ScanBMA [17]</p>
<ul class="simple">
<li><p>incorporates external information</p></li>
</ul>
<p>Inferelator [18]</p>
<ul class="simple">
<li><p>uses ODE</p></li>
<li><p>outputs a list of <strong>ordered</strong> gene pairs where each gene pair is associated with a positive, non-zero right <span class="math notranslate nohighlight">\(W(u, v) \in (0, 1]\)</span>. (score??)</p></li>
<li><p>gene pairs not shown in the output list are assumed to have a weight of <span class="math notranslate nohighlight">\(-1\)</span> (why not shown??)</p></li>
</ul>
</div>
<div class="section" id="supervised-methods">
<h3>Supervised Methods<a class="headerlink" href="#supervised-methods" title="Permalink to this headline">¶</a></h3>
<p>performance depends on the quality and the amount of available training data.</p>
<p>SVM [8,9,19,20].</p>
</div>
</div>
<div class="section" id="method">
<h2>Method<a class="headerlink" href="#method" title="Permalink to this headline">¶</a></h2>
<p>Overview</p>
<ul class="simple">
<li><p>use unsupervised method to infer a GRN, where links contains noise</p></li>
<li><p>sample positive amd negative links from this GRN, denoise and obtain cleaned data set</p></li>
<li><p>training supervised method based on the cleaned data set</p></li>
</ul>
<div class="section" id="feature-vector-construction">
<h3>Feature Vector Construction<a class="headerlink" href="#feature-vector-construction" title="Permalink to this headline">¶</a></h3>
<p>Given gene expression data, use an unsup method (Inferelator in this paper) to infer a GRN, which is a weighted directed graph <span class="math notranslate nohighlight">\(G=(V, E, W)\)</span></p>
<ul class="simple">
<li><p>node: gene</p></li>
<li><p><strong>directed</strong> link <span class="math notranslate nohighlight">\(e\)</span>: inferred regulatory relationship</p></li>
<li><p>weight: <span class="math notranslate nohighlight">\(W(e) \in (0,1]\)</span></p></li>
</ul>
<p>Traditional methods sample <span class="math notranslate nohighlight">\(E_{sample}\)</span> (both positive and negative) from <span class="math notranslate nohighlight">\(E\)</span>. And then use the gene expression data for gene nodes in the induced graph of set <span class="math notranslate nohighlight">\(E_{sample}\)</span>, to train supervised methods. But unsup methods often creates missing and spurious links in <span class="math notranslate nohighlight">\(E\)</span></p>
<ul class="simple">
<li><p>missing link: <span class="math notranslate nohighlight">\(e_m \notin G\)</span> but in ground truth GRN</p></li>
<li><p>spurious link: <span class="math notranslate nohighlight">\(e_s \in G\)</span> but not in ground truth GRN</p></li>
</ul>
<p>To improve this, we add a denoise step.</p>
</div>
<div class="section" id="data-clearning-algorithm">
<h3>Data Clearning Algorithm<a class="headerlink" href="#data-clearning-algorithm" title="Permalink to this headline">¶</a></h3>
<p>Input</p>
<ul class="simple">
<li><p>In <span class="math notranslate nohighlight">\(E_{sample}\)</span>, we have <span class="math notranslate nohighlight">\(m\)</span> links with the largest weights as positive samples, and <span class="math notranslate nohighlight">\(m\)</span> links with the smallest weights as negative samples.</p></li>
<li><p>Then, we construct feature vector for each (ordered??) pair of gene in <span class="math notranslate nohighlight">\(E_{sample}\)</span>, by simply concatenation of two gene expression vectors. For directed edge <span class="math notranslate nohighlight">\(e=(u, v) \in E_{sample}\)</span>, the feature vector is <span class="math notranslate nohighlight">\(\boldsymbol{x} _e=[\boldsymbol{g} _u; \boldsymbol{g} _v] \in \mathbb{R} ^{2p}\)</span>.</p></li>
<li><p>The feature matrix is then <span class="math notranslate nohighlight">\(\boldsymbol{X} \in \mathbb{R} ^{2m \times 2p}\)</span>.</p></li>
</ul>
<p>Algorithm</p>
<ol>
<li><p>calculate a distance matrix <span class="math notranslate nohighlight">\(\boldsymbol{D} \in \mathbb{R} ^{2m \times 2m}\)</span> for the feature vectors by Laplacian kernel</p>
<div class="math notranslate nohighlight">
\[
    D_{ij} = d(\boldsymbol{x}_i, \boldsymbol{x} _j) = \exp (-\sigma \left\| \boldsymbol{x} _i - \boldsymbol{x} _j \right\|_2 )
    \]</div>
<p>where <span class="math notranslate nohighlight">\(\sigma=1\)</span> in this paper.</p>
</li>
<li><p>project the training set onto the eigenvectors of the distance matrix, to obtain noise-removed features (why noise is removed/why this works?! see [35])</p>
<ul>
<li><p>EVD: <span class="math notranslate nohighlight">\(\boldsymbol{D} = \boldsymbol{V} \boldsymbol{\Sigma} \boldsymbol{V}^{\top}\)</span></p></li>
<li><p>Use the smallest <span class="math notranslate nohighlight">\(t\)</span> eigenvectors, denoted <span class="math notranslate nohighlight">\(\boldsymbol{V} _t\)</span></p></li>
<li><p>project data matrix <span class="math notranslate nohighlight">\(\boldsymbol{X}\)</span> onto <span class="math notranslate nohighlight">\(\boldsymbol{V} _t\)</span>:</p>
<div class="math notranslate nohighlight">
\[\boldsymbol{C} = \boldsymbol{V} _t \boldsymbol{V} _t ^{\top}\boldsymbol{X} \in \mathbb{R} ^{2m\times 2p}\]</div>
<p>each column of <span class="math notranslate nohighlight">\(\boldsymbol{C}\)</span> is a noise-removed feature vector.</p>
</li>
</ul>
</li>
<li><p>select important features from the noise-removed features, which then form a <strong>cleaned</strong> training set to be used to train supervised methods (SVM and NN in this paper)</p>
<p>Specifically, compute column sum of <span class="math notranslate nohighlight">\(\boldsymbol{C}\)</span>, find the <span class="math notranslate nohighlight">\(k\)</span> minimum values from <span class="math notranslate nohighlight">\(2p\)</span> values (why minimum ??). Return these <span class="math notranslate nohighlight">\(k\)</span> noise-removed feature vectors, denoted <span class="math notranslate nohighlight">\(\bar{\boldsymbol{C}} \in \mathbb{R} ^{2m \times k}\)</span>, termed ‘important features’, forming the cleaned training set. (in this paper <span class="math notranslate nohighlight">\(k=10\)</span>).</p>
</li>
</ol>
</div>
</div>
<div class="section" id="experiments">
<h2>Experiments<a class="headerlink" href="#experiments" title="Permalink to this headline">¶</a></h2>
<div class="section" id="data">
<h3>Data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h3>
<p>Three datasets in DREAM4 100-gene in silicon [13, 14].</p>
<ul class="simple">
<li><p>Each dataset contains 100 genes</p>
<ul>
<li><p>For each gene, there are 10 times series, each time seires has 21 time points. Hence <span class="math notranslate nohighlight">\(\boldsymbol{g} \in \mathbb{R} ^{210}\)</span> and <span class="math notranslate nohighlight">\(\boldsymbol{x} \in \mathbb{R} ^{420}\)</span>.</p></li>
</ul>
</li>
<li><p>Each dataset is associated with a gold standard file, representing the true regulatory relationships (directed) between two genes.</p></li>
</ul>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p>Net1</p></th>
<th class="head"><p>Net2</p></th>
<th class="head"><p>Net3</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Directed</p></td>
<td><p>Yes</p></td>
<td><p>Yes</p></td>
<td><p>Yes</p></td>
</tr>
<tr class="row-odd"><td><p>Nodes</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-even"><td><p>True present links</p></td>
<td><p>176</p></td>
<td><p>249</p></td>
<td><p>195</p></td>
</tr>
<tr class="row-odd"><td><p>True missing links</p></td>
<td><p>9724</p></td>
<td><p>9651</p></td>
<td><p>9705</p></td>
</tr>
</tbody>
</table>
<div class="section" id="training-set">
<h4>Training set<a class="headerlink" href="#training-set" title="Permalink to this headline">¶</a></h4>
<p>For each dataset,</p>
<ol class="simple">
<li><p>run Inferetor to obtain a list of ordered gene pairs with weights</p></li>
<li><p>select</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(m\)</span> gene pairs with largest weights as positive examples</p></li>
<li><p><span class="math notranslate nohighlight">\(m\)</span> gene pairs with smallest weights as positive examples</p></li>
</ul>
</li>
<li><p>construct</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(2m\)</span> feture vectors, as original training set <span class="math notranslate nohighlight">\(\boldsymbol{X}_{train} \in \mathbb{R} ^{2m \times 2p}\)</span>.</p></li>
<li><p><span class="math notranslate nohighlight">\(2m\)</span> labels, including <span class="math notranslate nohighlight">\(m\)</span> number of 1’s and <span class="math notranslate nohighlight">\(m\)</span> number of -1’s, denoted <span class="math notranslate nohighlight">\(\boldsymbol{y} \in \mathbb{R} ^{2m}\)</span></p></li>
</ul>
</li>
</ol>
</div>
<div class="section" id="test-set">
<h4>Test set<a class="headerlink" href="#test-set" title="Permalink to this headline">¶</a></h4>
<p>For each dataset, we generated three sets of <strong>testing data</strong> <span class="math notranslate nohighlight">\(\boldsymbol{X} _{test_1}, \boldsymbol{X} _{test_2}, \boldsymbol{X} _{test_3}\)</span></p>
<ul class="simple">
<li><p>Each testing set contains 50 randomly selected links (gene pairs) from the gold standard.</p>
<ul>
<li><p>Among the 50 links, 25 are true present links and 25 are true missing links in the gold standard.</p></li>
</ul>
</li>
<li><p>9 testing sets totally</p></li>
<li><p>disjoint with <span class="math notranslate nohighlight">\(2m\)</span> gene pairs in training set</p></li>
</ul>
</div>
</div>
<div class="section" id="pipeline">
<h3>Pipeline<a class="headerlink" href="#pipeline" title="Permalink to this headline">¶</a></h3>
<p>For each dataset,</p>
<ol class="simple">
<li><p>Construct one training set <span class="math notranslate nohighlight">\(\boldsymbol{X} _{train}\)</span> and three test sets <span class="math notranslate nohighlight">\(\boldsymbol{X} _{test_1}, \boldsymbol{X} _{test_2}, \boldsymbol{X} _{test_3}\)</span> as described above.</p></li>
<li><p>Run proposed data cleaning algorithm on <span class="math notranslate nohighlight">\(\boldsymbol{X} _{train}\)</span> to obtain cleaned training set <span class="math notranslate nohighlight">\(\bar{\boldsymbol{C}} \in \mathbb{R} ^{2m \times k}\)</span>, with label <span class="math notranslate nohighlight">\(\boldsymbol{y}\)</span>.</p></li>
<li><p>Train classifiers (SVM, DNN) on <span class="math notranslate nohighlight">\(\bar{\boldsymbol{C}}\)</span>.</p></li>
<li><p>Evaluate trained classifiers on <span class="math notranslate nohighlight">\(\boldsymbol{X} _{test_1}, \boldsymbol{X} _{test_2}, \boldsymbol{X} _{test_3}\)</span>.</p></li>
</ol>
</div>
<div class="section" id="evaluation">
<h3>Evaluation<a class="headerlink" href="#evaluation" title="Permalink to this headline">¶</a></h3>
<p>Consider three supervised classification algorithms</p>
<ul class="simple">
<li><p>SVM with various kernels</p></li>
<li><p>DNN with weights initialized by deep belief networks (DNN_DBN)</p></li>
<li><p>DNN with weights initialized by stack AutoEncoder (DNN_Auto)</p></li>
</ul>
<p>Option of probability estimation was turned on in each algorithm.</p>
<p>Classification rule:</p>
<ul class="simple">
<li><p>SVM, DNN:</p>
<ul>
<li><p>positive: prob &gt; median prob</p></li>
<li><p>negative: prob &lt; median prob</p></li>
</ul>
</li>
<li><p>Inferelator (unsupervised method on test sets)</p>
<ul>
<li><p>positive: weight &gt; median weight</p></li>
<li><p>negative: weight &lt; median weight</p></li>
</ul>
</li>
</ul>
<p>Use averaged AUC across 9 testing sets as creiteria for comparison. (weird definition of AUC in eq.11)??</p>
</div>
<div class="section" id="results">
<h3>Results<a class="headerlink" href="#results" title="Permalink to this headline">¶</a></h3>
<p>Compare SVM with various kernel (Fig.1)</p>
<ul class="simple">
<li><p>Linear kernel perfomrs the best among Gaussian, sigmoid, polynomial kernels.</p></li>
</ul>
<p>Compare SVM with linear kernel, DBN, Inf (TABLE II)</p>
<ul class="simple">
<li><p>Net1: SVM_L, Net2: DNN_Auto, Net3: DNN_DBN.</p></li>
<li><p>All three supervised methods perform better than the unsupervised method Inferelator.</p></li>
<li><p>Overall, SVM_L performs better than DNN.</p>
<ul>
<li><p>Possibly because this is a <strong>binary</strong> classification with <strong>small</strong> dataset.</p></li>
</ul>
</li>
</ul>
<p>Compare results w/ and w/o the proposed cleaning algorithm (Fig.2)</p>
<ul class="simple">
<li><p>AUC are smaller for SVM_L and DNNs.</p>
<ul>
<li><p>SVM_L suffers the most when data is not cleaned.</p></li>
</ul>
</li>
<li><p>showing the effectiveness of our proposed data cleaning algorithm.</p></li>
</ul>
<p>Compare different important feature dimension <span class="math notranslate nohighlight">\(k=20, 30, 40\)</span> (Fig.3)</p>
<ul class="simple">
<li><p>SVM_L still the best for all <span class="math notranslate nohighlight">\(k\)</span></p></li>
</ul>
<p>Varying other hyperparameters give similar results.</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(m\)</span>: #sampled gene pairs</p></li>
<li><p><span class="math notranslate nohighlight">\(\sigma\)</span>: paramter in Laplacian kernel</p></li>
<li><p><span class="math notranslate nohighlight">\(t\)</span>: #eigenvectors</p></li>
</ul>
</div>
</div>
<div class="section" id="takeaway">
<h2>Takeaway<a class="headerlink" href="#takeaway" title="Permalink to this headline">¶</a></h2>
<div class="section" id="ref-of-interest">
<h3>Ref of interest<a class="headerlink" href="#ref-of-interest" title="Permalink to this headline">¶</a></h3>
<p>Inferelator [18] [<a class="reference external" href="https://genomebiology.biomedcentral.com/track/pdf/10.1186/gb-2006-7-5-r36.pdf">PDF</a>]</p>
<ul class="simple">
<li><p>why output <strong>ordered</strong> pairs??</p></li>
</ul>
<p>Related noise-filtering algorithm by Ouyang et al. [35] [<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4720285/pdf/pone.0146925.pdf">PDF</a>]</p>
<ul class="simple">
<li><p>Ouyang’s algorithm is designed for undirected networks, and employs EVD of Laplacian matrix</p></li>
<li><p>In this paper, we consider directed networks (output of Inferelator are <strong>ordered</strong> gene pairs), hence Laplacian matrix is not symmetric, whose EVD involves complex numbers</p></li>
<li><p>To solve this, use kernel function to construct a distance matrix <span class="math notranslate nohighlight">\(\boldsymbol{D}\)</span>.</p></li>
</ul>
</div>
<div class="section" id="questions">
<h3>Questions<a class="headerlink" href="#questions" title="Permalink to this headline">¶</a></h3>
<p>Robustness to order of genes</p>
<ul class="simple">
<li><p>For <span class="math notranslate nohighlight">\(m\)</span> negative examples in training set, if we reverse the order (since their order does not matter), how will that affect the <span class="math notranslate nohighlight">\(k\)</span> import feture vector?</p></li>
<li><p>How about reversing order for <span class="math notranslate nohighlight">\(m\)</span> positive orders? If we only care about interaction, but not care direction, then their order does not matter.</p></li>
</ul>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./summary"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="31-grn-ssu.html" title="previous page">Sup, semi-sup, unsup inference</a>
    <a class='right-next' id="next-link" href="71-grgnn.html" title="next page">GRGNN</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Dennis Zheng<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>