
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>GRGNN &#8212; Inference for Gene Regulartory Networks</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.e8e5499552300ddf5d7adccae7cc3b70.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/style.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-svg.js"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="For plants" href="71-plants.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      
      <h1 class="site-logo" id="site-title">Inference for Gene Regulartory Networks</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../index.html">
   Inference for Gene Regulartory Networks
  </a>
 </li>
</ul>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../intro/00-intro.html">
   Introduction
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../intro/11-grn.html">
     Gene Regulatory Networks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../intro/31-datasets.html">
     Datasets
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="00-summary.html">
   Paper Summary
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="11-bio-net-dl.html">
     Biological networks and deep learning
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="31-grn-ssu.html">
     Sup, semi-sup, unsup inference
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="53-grn-denoise.html">
     A Denoise Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="61-grn-ggm.html">
     Compare RN GNN BN
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="71-plants.html">
     For plants
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     GRGNN
    </a>
   </li>
  </ul>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/summary/81-grgnn.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#review-methods-and-limitation">
   Review Methods and Limitation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#model">
   Model
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#construct-noisy-skeletons">
     Construct noisy skeletons
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#extract-enclosed-subgraphs">
     Extract enclosed subgraphs
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#add-node-labels-and-features">
     Add node labels and features
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#build-ensemble-gnn-classifiers">
     Build ensemble GNN classifiers
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experiments">
   Experiments
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#supervised-transductive">
     Supervised + Transductive
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#supervised-inductive">
     Supervised + Inductive
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#unsupervised">
     Unsupervised
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#number-of-layers">
     Number of Layers
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#heuristic-functions">
     Heuristic Functions
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#takeaway">
   Takeaway
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#limitation">
     Limitation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#ref">
     Ref
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#qs">
     Qs
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#for-writing">
     For writing
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="grgnn">
<h1>GRGNN<a class="headerlink" href="#grgnn" title="Permalink to this headline">¶</a></h1>
<p>[<a class="reference external" href="https://nbviewer.jupyter.org/github/dennissxz/gene-regulatory-networks/blob/main/reading/grn-grgnn.pdf">PDF</a>] [<a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S200103702030444X?via%3Dihub">DOI</a>] [<a class="reference external" href="https://github.com/juexinwang/GRGNN">Github</a>] Wang, Juexin, Anjun Ma, Qin Ma, Dong Xu, and Trupti Joshi. “Inductive inference of gene regulatory network using supervised and semi-supervised graph neural networks.” Computational and Structural Biotechnology Journal 18 (2020): 3335-3343.</p>
<p>We propose an end-to-end gene regulatory graph neural network (GRGNN) approach to reconstruct GRNs from scratch utilizing the gene expression data, in both a supervised and a semisupervised framework.</p>
<p>highlight: !! link prediction to graph classification</p>
<ul class="simple">
<li><p>To get better inductive generalization capability, GRN inference is formulated as a graph classification problem, to distinguish whether a <strong>subgraph</strong> <strong>centered</strong> at two nodes contains the link between the two nodes</p></li>
<li><p>A linked pair between a transcription factor (TF) and a target gene, and their neighbors are labeled as a positive subgraph, while an unlinked TF and target gene pair and their neighbors are labeled as a negative subgraph.</p></li>
</ul>
<p>Model intro</p>
<ul class="simple">
<li><p>A GNN model is constructed with node features from both explicit gene expression and graph embedding.</p></li>
<li><p>We demonstrate a noisy starting graph structure built from partial information (i.e. noisy skeleton) based on heuristics on gene expression data, such as Pearson’s correlation coefficient and mutual information which can help guide the GRN inference through an appropriate <strong>ensemble</strong> technique.</p></li>
</ul>
<p>To the best of our knowledge, this is the first work to infer gene regulatory networks through graph neural networks. Our contributions in this paper are</p>
<ol class="simple">
<li><p>introduction of a supervised/semisupervised graph classification framework for gene regulatory network inference,</p></li>
<li><p>using noisy starting skeleton to guide link prediction in the graph,</p></li>
<li><p>efforts of <strong>inductive</strong> inference GRN across unknown species and conditions (?? GRNs can be constructed with the same input as the unsupervised methods without using new labels in the new condition or species)</p></li>
</ol>
<p>Dataset: DREAM5 GRN inference benchmarks.</p>
<ul class="simple">
<li><p>Take E. coli for example, there are 334 TF and 4177 Targets, in total 344*(344+4177-1)=1,506,340 possible links in total, and only 2,066 among them are confirmed gold standard positive links</p></li>
</ul>
<div class="section" id="review-methods-and-limitation">
<h2>Review Methods and Limitation<a class="headerlink" href="#review-methods-and-limitation" title="Permalink to this headline">¶</a></h2>
<p>Unsupervised Methods</p>
<ul class="simple">
<li><p>regression-based (TFs are selected by target gene through sparse linear-regression)</p>
<ul>
<li><p>TIGRESS [14]</p></li>
</ul>
</li>
<li><p>information-based methods (ranking edges based on variants of mutual informaiton)</p>
<ul>
<li><p>ARACNE</p></li>
<li><p>CLR</p></li>
<li><p>MRNET</p></li>
</ul>
</li>
<li><p>correlation-basd methods</p></li>
<li><p>Bayesian networks by optimizing posterior prob using different heuristic searches [1]</p></li>
<li><p>GENIE3 [15] is a well-established and widely accepted method based on ensemble random forest regression of gene expression levels between TF and targets. Obtained best performance among all the methods at that time.</p></li>
</ul>
<p>Supervised methods and limitations</p>
<ul class="simple">
<li><p>Because of heterozygous data sources, these supervised methods usually have limited generalizable capabilities in complex biological mechanisms.</p></li>
<li><p>For a practical GRN inferring problem, there is usually no known relationship ready for training, which makes it unfeasible to predict gene regulatory relationships inductively in practice.</p></li>
</ul>
<p>Motivation: Motif</p>
<ul class="simple">
<li><p>Gene regulatory activities always act as a whole system with a set of genes to perform a biological function [19].</p></li>
<li><p>Network motif [2] is a widely accepted biological hypothesis,</p></li>
<li><p>Existing supervised GRN inferring methods usually only take the two endpoints of the regulatory interactions as the input, and then treat these known TF/target gene interactions independently in the training processes, and hence neglect the <strong>global</strong> relationships among these interactions.</p></li>
</ul>
</div>
<div class="section" id="model">
<h2>Model<a class="headerlink" href="#model" title="Permalink to this headline">¶</a></h2>
<p>Hypothesis</p>
<ul class="simple">
<li><p>the features of two nodes and their neighbors (local structure) can decide whether they form a TF and target gene pair, which is consistent with the network motif hypothesis.</p></li>
</ul>
<div class="section" id="construct-noisy-skeletons">
<h3>Construct noisy skeletons<a class="headerlink" href="#construct-noisy-skeletons" title="Permalink to this headline">¶</a></h3>
<div class="math notranslate nohighlight">
\[\begin{split}
A_{t, g^{\prime}}=\left\{\begin{array}{ccc}
1 &amp; \text { if } &amp; H_{i}(t, g)&gt;=\text { threshold }_{i} \\
0 &amp; &amp; \text { otherwise }
\end{array}\right.
\end{split}\]</div>
<ul class="simple">
<li><p>use heuristic methods <span class="math notranslate nohighlight">\(H(t, g)\)</span>, e.g. correlation, MI</p></li>
<li><p>threshold are set as the parameters for training</p></li>
</ul>
<p>These inferred links are noisy, but integrating these links as a starting skeleton can guide training.</p>
</div>
<div class="section" id="extract-enclosed-subgraphs">
<h3>Extract enclosed subgraphs<a class="headerlink" href="#extract-enclosed-subgraphs" title="Permalink to this headline">¶</a></h3>
<p>For each known <span class="math notranslate nohighlight">\((t, g )\in E\)</span></p>
<ul class="simple">
<li><p>extract positive subgrah <span class="math notranslate nohighlight">\(SG_i(t, g) ^+\)</span> from noisy skeleton <span class="math notranslate nohighlight">\(i\)</span></p>
<ul>
<li><p>include <span class="math notranslate nohighlight">\(h\)</span>-hop neighbors on noisy skeleton</p></li>
<li><p>choice of <span class="math notranslate nohighlight">\(h\)</span> depend on complexity of data</p></li>
</ul>
</li>
</ul>
<p>Randomly select <span class="math notranslate nohighlight">\((t, g )\notin E\)</span></p>
<ul class="simple">
<li><p>extract negative subgraph <span class="math notranslate nohighlight">\(SG(t, g) ^-\)</span> from noisy skeleton <span class="math notranslate nohighlight">\(i\)</span></p>
<ul>
<li><p>include <span class="math notranslate nohighlight">\(h\)</span>-hop neighbors on noisy skeleton</p></li>
</ul>
</li>
<li><p>may select false negatives due to undiscovered regulatory relationship, but common practice for choosing negative examples</p></li>
</ul>
<p>To get a balanced dataset, set same size. A semi-supervised learning strategy is introduced to select a Reliable Negative sample set from the unlabeled datasets. See 2.5.</p>
</div>
<div class="section" id="add-node-labels-and-features">
<h3>Add node labels and features<a class="headerlink" href="#add-node-labels-and-features" title="Permalink to this headline">¶</a></h3>
<p>A perfect hashing labelling [39] function <span class="math notranslate nohighlight">\(label(i)\)</span> is used for marking node <span class="math notranslate nohighlight">\(i\)</span>’ s roles in <span class="math notranslate nohighlight">\(SG(t, g)\)</span></p>
<div class="math notranslate nohighlight">
\[\begin{split}
\text { label }(i)=\left\{\begin{array}{cc}
1 &amp; \text { if } \quad i=t \mid g \\
1+\min \left(d_{t}+d_{g}+(d / 2)[(d / 2)+(d \% 2)-1]\right) &amp; \text { otherwise }
\end{array}\right.
\end{split}\]</div>
<p>where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(d_i = d(i, t)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(d_g = d(i, g)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(d = d_t + d_g\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\((d/2)\)</span> and <span class="math notranslate nohighlight">\(d \%2\)</span> are the integer quotient and remainder of <span class="math notranslate nohighlight">\(d\)</span> divided by <span class="math notranslate nohighlight">\(2\)</span>.</p></li>
<li><p>min (??)</p></li>
</ul>
<p>why add node labels??</p>
<p>Only the centered target nodes <span class="math notranslate nohighlight">\(t\)</span> and <span class="math notranslate nohighlight">\(g\)</span> are labeled with 1, while the importance of nodes decreases when the node is far away from the center nodes. With appropriate labels, GNN can learn the structural information whether a link exists between the target nodes.</p>
<p>Node features include</p>
<ul class="simple">
<li><p><strong>explicit features</strong> computed from gene expression data</p>
<ul>
<li><p>use <span class="math notranslate nohighlight">\(z\)</span>-score [6] and summary statistics: sd, four quantile percentages</p></li>
</ul>
</li>
<li><p><strong>structural embeddings</strong>, e.g node2vec, from this noisy skeleton <span class="math notranslate nohighlight">\(GRN_i ^\prime\)</span>.</p></li>
</ul>
</div>
<div class="section" id="build-ensemble-gnn-classifiers">
<h3>Build ensemble GNN classifiers<a class="headerlink" href="#build-ensemble-gnn-classifiers" title="Permalink to this headline">¶</a></h3>
<p>DGCNN [40] is used for graph classification, which adopts a quasi Weisfeiler-Lehman subtree model [34] to extract nodes’ local substructure features, and pool these nodes in order.</p>
<p>Finally, a convolutional network (CNN) follows to read sorted (node label used here??) graph representations and make predictions.</p>
<p>Ensemble method combines prediction from each GNN that is built for each heuristic function. The weights can be trained either through a neural network or a simple least squre regression. This paper demonstrates a <span class="math notranslate nohighlight">\(k=2\)</span> case and simply manually set equal weights <span class="math notranslate nohighlight">\((0.5, 0.5)\)</span>.</p>
</div>
</div>
<div class="section" id="experiments">
<h2>Experiments<a class="headerlink" href="#experiments" title="Permalink to this headline">¶</a></h2>
<p>Use E. coli and S. cerevisiae in DREAM5 challenge.</p>
<div class="section" id="supervised-transductive">
<h3>Supervised + Transductive<a class="headerlink" href="#supervised-transductive" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>Task: predict unknown given parts of the known in the same species.</p></li>
<li><p>Evaluation: 3-fold cross-validation.</p></li>
<li><p>baseline: SVM, RF</p></li>
<li><p><span class="math notranslate nohighlight">\(h=1\)</span>, explicit feature dim=1.</p></li>
</ul>
<p>Ensembled GRGNN performs better or at least the same with SVM/RF in GRN inferences in nearly all the criteria.</p>
</div>
<div class="section" id="supervised-inductive">
<h3>Supervised + Inductive<a class="headerlink" href="#supervised-inductive" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>Task: The model trained from E. coli was applied to predict regulatory relationships in S. cerevisiae, and the model trained from S. cerevisiae was used to predict E. coli.</p></li>
</ul>
<p>Findings:</p>
<ul class="simple">
<li><p>Ensembled GRGNN outperforms SVM/RF in GRN inferences in nearly all the criteria.</p></li>
<li><p>1-hop GRGNN outperforms 0-hop GRGNN persistently in most of the criteria of both datasets, , which indicates integrating neighbors brings more predictive power to the graph model.</p></li>
<li><p>adding graph embedding of the enclosed subgraph generally improves the performances for GRGNN.</p></li>
</ul>
</div>
<div class="section" id="unsupervised">
<h3>Unsupervised<a class="headerlink" href="#unsupervised" title="Permalink to this headline">¶</a></h3>
<p>GRGNN inductive compared with unsupervised method.</p>
<p>GRGNN outperforms all other methods on both ROC curve and Precision-Recall curve.</p>
</div>
<div class="section" id="number-of-layers">
<h3>Number of Layers<a class="headerlink" href="#number-of-layers" title="Permalink to this headline">¶</a></h3>
<p>How to choose <span class="math notranslate nohighlight">\(d\)</span> in <span class="math notranslate nohighlight">\(d\)</span>-hop GRGNN? Experiment <span class="math notranslate nohighlight">\(h=0\)</span> to 9.</p>
<p>Findings:</p>
<ul class="simple">
<li><p>adding 1- hop to 0-hop brings extra predictive power with the neighbors as the local structure in the graph. After that, adding more hops does
not seem to bring significantly better results in GRN.</p></li>
<li><p>This phenomenon may indicate that few hops of GNN contain almost all information for link prediction from its local structure in the graph,</p></li>
</ul>
</div>
<div class="section" id="heuristic-functions">
<h3>Heuristic Functions<a class="headerlink" href="#heuristic-functions" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>compared with random skeleton, better performance (3.6)</p></li>
<li><p>robust to threshold. (3.7)</p></li>
</ul>
</div>
</div>
<div class="section" id="takeaway">
<h2>Takeaway<a class="headerlink" href="#takeaway" title="Permalink to this headline">¶</a></h2>
<div class="section" id="limitation">
<h3>Limitation<a class="headerlink" href="#limitation" title="Permalink to this headline">¶</a></h3>
<p>The main limitation of this work is the datasets used. E. coli and S. cerevisiae are relatively well-studied small model species. These data are the <strong>only benchmark</strong> having systematically clear, experimental validated, gold standard regulatory relationships.</p>
</div>
<div class="section" id="ref">
<h3>Ref<a class="headerlink" href="#ref" title="Permalink to this headline">¶</a></h3>
<p>[39] SEAL: Zhang Muhan, Chen Yixin. Link prediction based on graph neural networks. Advances in Neural Information Processing Systems 31. 2018. p. 5165–75. http://papers.nips.cc/paper/7763-linkprediction-based-on-graph-neural-networks.pdf.</p>
<p>[40] DGCNN: Zhang Muhan, Cui Zhicheng, Neumann Marion, Chen Yixin. An end-to-end deep learning architecture for graph classification. Thirty-Second AAAI Conference on Artificial Intelligence, 2018. https://www.aaai.org/ocs/index.php/AAAI/AAAI18/paper/view/17146.</p>
<p>[15] GENIE: Huynh-Thu V, Irrthum A, Wehenkel L, et al. Inferring regulatory networks from expression data using tree-based methods. PLoS One 2010;5(9):e12776.</p>
<p>[6] z-score</p>
</div>
<div class="section" id="qs">
<h3>Qs<a class="headerlink" href="#qs" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>How to ‘center’ two nodes in a graph in GNN for graph classification?</p></li>
<li><p>z-score <span class="math notranslate nohighlight">\(R_i\)</span>?</p></li>
</ul>
</div>
<div class="section" id="for-writing">
<h3>For writing<a class="headerlink" href="#for-writing" title="Permalink to this headline">¶</a></h3>
<p>introduction</p>
<ul class="simple">
<li><p>necessity of GRN.</p></li>
<li><p>classical methods</p></li>
</ul>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./summary"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="71-plants.html" title="previous page">For plants</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Dennis Zheng<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>